### PR TITLE
CNS-734: use antehandler to mark redelegation

### PR DESCRIPTION
**Pull request by Yaroms**

### CODE REVIEW

File: x/dualstaking/ante/ante_handler.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/lavanet/lava/x/dualstaking/keeper"
+)
+
+// RedelegationFlager sets the GasMeter in the Context and wraps the next AnteHandler with a defer clause
```
Comment (omerlavanet) [inline x/dualstaking/ante/ante_handler.go:12]: this comment is wrong

File: x/dualstaking/ante/ante_handler.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+// to recover from any downstream OutOfGas panics in the AnteHandler chain to return an error with information
+// on gas provided and gas used.
+// CONTRACT: Must be first decorator in the chain
```
Comment (omerlavanet) [inline x/dualstaking/ante/ante_handler.go:15]: this one also

File: x/dualstaking/keeper/hooks.go [hunk 3]
```diff
@@ -46,6 +46,10 @@ func (h Hooks) BeforeDelegationSharesModified(ctx sdk.Context, delAddr sdk.AccAd
 // create new delegation period record
 // add description
 func (h Hooks) AfterDelegationModified(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress) error {
+	if RedelegationFlag {
```
Comment (omerlavanet) [inline x/dualstaking/keeper/hooks.go:49]: can we call this flag with a flag that explains what it does, something like: disableDualstakingHook or something

**Review by omerlavanet — CHANGES_REQUESTED** [review]

File: x/dualstaking/keeper/hooks.go (hunk 3 above)
Comment (Yaroms) [reply x/dualstaking/keeper/hooks.go:49]: done

**Review by omerlavanet — DISMISSED** [review]

**Review by omerlavanet — APPROVED** [review]

**Pull request merged**